packages/core/src/external/aws/lambda-logic/bundle-to-html-bmi.ts

```diff
@@ -968,6 +963,27 @@ function createWeightComoborbidities(
   return createSection("Weight-related Comorbidities", conditionTableContents);
 }
 
+function getLocationFromEncounters(
+  conditionId: string | undefined,
+  encounters: Encounter[],
+  locations: Location[]
+): Location | undefined {
+  if (!conditionId) return undefined;
+
+  const encounter = encounters.find(enc =>
+    enc.diagnosis?.some(diag => diag.condition?.reference?.includes(conditionId))
+  );
+
+  if (!encounter) return undefined;
+
+  const locationReference = encounter.location?.[0]?.location?.reference;
+  if (!locationReference) return undefined;
+
+  const locationId = locationReference.split("/")[1];
+
+  return locations.find(location => location.id === locationId);
+}
```

Comment:
this name seems potentially misleading based on the code - whats this function trying to do?

Reply:
this looks like `getLocationOfDiagnosis` or something.